torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1276,13 +1277,46 @@ ProcessGroupNCCL::~ProcessGroupNCCL() {
 #endif
 }
 
-void dump_debugging_info() {
+void storeTraceInfoToLocalDisk(int rank, const std::string& ncclTrace) {
+  const char* fileName = parseEnvVarString(
+      "TORCH_NCCL_DEBUG_INFO_TEMP_FILE", "/tmp/nccl_trace_rank_");
+  auto filename = c10::str(std::string(fileName), rank);
+
+  // Open a file for writing. The ios::binary flag is used to write data as
+  // binary.
+  std::ofstream file(filename, std::ios::binary);
+
+  // Check if the file was opened successfully.
+  if (!file.is_open()) {
+    LOG(ERROR) << "Error opening file for writing NCCLPG debug info: " << filename;
+    return;
+  }
+
+  // Write the size of the string followed by the string itself.
+  // This can help in retrieving the strings back if needed.
+  size_t size = ncclTrace.size();
+  file.write(reinterpret_cast<const char*>(&size), sizeof(size));
+  file.write(ncclTrace.data(), size);
```

Comment:
we might want to use zip writer here. i suppose that could be an optimization later, but how large are these files anyway?

Reply:
Well I guess it depends on how big we want to set for `TORCH_NCCL_TRACE_BUFFER_SIZE`? In the scenario of 16/32K training, for a typical LLM transformer, each layer will use about 2-4 collectives for FSDP, 4 for TP/SP and 0-2 for PP. So 10 collectives at most for one layer, and if we think looking back 3-5 layers should be suffice, then we probably only need 30-50 entries. If one entry take about 1KB(just random guess, maybe @zdevito can correct me if I am wrong). Then it would be 30-50 KB?